internal/builders/golang/build.go

```diff
@@ -266,6 +275,10 @@ func buildGoBuildLine(ctx *context.Context, build config.Build, details config.B
 		cmd = append(cmd, "-ldflags="+strings.Join(ldflags, " "))
 	}
 
+	if details.Buildmode != "" {
+		cmd = append(cmd, "-buildmode="+details.Buildmode)
```

Comment:
I think it would be good to check if cmd already has a `-buildmode`, and if so, error out... 

my though behind that: people might set different buildmodes (one via flags and another with the buildmode option introduced in this pr), and it could get a bit confusing.

a warning might be good enough too, though.

wdyt?

Reply:
While testing this, I realised the same could happen with ldflags and tags, which I guess could cause some problems too? I hope you don't mind I pushed a fix for it in this same MR, just for consistency.
As you suggested, I made it into a plain warning because erroring could be a breaking change.

Reply:
ah, indeed, that looks nice, thanks!